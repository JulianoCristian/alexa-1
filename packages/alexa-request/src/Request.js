import { Request as Constants } from 'alexa-constants';

export default class Request {
  static session = (...args) => new Request().session(...args);
  static intent = (...args) => new Request().intent(...args);
  static launchRequest = (...args) => new Request().launchRequest(...args);
  static sessionEndedRequest = (...args) => new Request().sessionEndedRequest(...args);

  constructor(state = {}) {
    this.state = state;
  }

  session(session) {
    return new Request({
      ...this.state,
      session: {
        ...this.state.session,
        ...session
      }
    });
  }

  intent(name, slots = {}) {
    const slotData = Object.entries(slots).reduce((state, [name, value]) => ({ ...state, [name]: { name, value } }), {});
    return new Request({
      ...this.state,
      request: {
        type: Constants.Intent,
        intent: {
          name: name,
          ...(Object.keys(slotData).length ? { slots: slotData } : null)
        }
      }
    });
  }

  launchRequest() {
    return new Request({
      ...this.state,
      request: {
        type: Constants.Launch
      }
    });
  }

  sessionEndedRequest() {
    return new Request({
      ...this.state,
      request: {
        type: Constants.SessionEnded
      }
    });
  }

  build() {
    return { ...this.state };
  }
}
