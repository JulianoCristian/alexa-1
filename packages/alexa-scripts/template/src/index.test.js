import Skill from './index';
import Request from 'alexa-request';

test('Launch intent', () => {
  const event = Request.launchRequest().build();
  return expect(Skill(event)).resolves.toMatchSnapshot();
});

test('Hello intent', () => {
  const event = Request.intent('hello', { name: 'world' }).build();
  return expect(Skill(event)).resolves.toMatchSnapshot();
});

test('Help intent', () => {
  const event = Request.intent('AMAZON.HelpIntent').build();
  return expect(Skill(event)).resolves.toMatchSnapshot();
});

test('Stop intent', () => {
  const event = Request.intent('AMAZON.StopIntent').build();
  return expect(Skill(event)).resolves.toMatchSnapshot();
});

test('Cancel intent', () => {
  const event = Request.intent('AMAZON.CancelIntent').build();
  return expect(Skill(event)).resolves.toMatchSnapshot();
});
