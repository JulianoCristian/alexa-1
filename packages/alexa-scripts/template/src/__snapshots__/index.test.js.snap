// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cancel intent 1`] = `
Object {
  "response": Object {
    "outputSpeech": Object {
      "ssml": "<speak>Goodbye!</speak>",
      "type": "SSML",
    },
    "shouldEndSession": true,
  },
  "version": "1.0",
}
`;

exports[`Hello intent 1`] = `
Object {
  "response": Object {
    "card": Object {
      "content": "Hello world!",
      "title": "Hello toâ€¦",
      "type": "Simple",
    },
    "outputSpeech": Object {
      "ssml": "<speak>Hello world!</speak>",
      "type": "SSML",
    },
    "shouldEndSession": true,
  },
  "version": "1.0",
}
`;

exports[`Help intent 1`] = `
Object {
  "response": Object {
    "outputSpeech": Object {
      "text": "I say hello to people. Who should I say hello to?",
      "type": "PlainText",
    },
    "reprompt": Object {
      "outputSpeech": Object {
        "text": "Who should I say hello to?",
        "type": "PlainText",
      },
    },
    "shouldEndSession": false,
  },
  "version": "1.0",
}
`;

exports[`Launch intent 1`] = `
Object {
  "response": Object {
    "outputSpeech": Object {
      "text": "Hello World Alexa skill launched! Who would you like to say hello to?",
      "type": "PlainText",
    },
    "reprompt": Object {
      "outputSpeech": Object {
        "text": "Who would you like to say hello to?",
        "type": "PlainText",
      },
    },
    "shouldEndSession": false,
  },
  "version": "1.0",
}
`;

exports[`Stop intent 1`] = `
Object {
  "response": Object {
    "outputSpeech": Object {
      "ssml": "<speak>Goodbye!</speak>",
      "type": "SSML",
    },
    "shouldEndSession": true,
  },
  "version": "1.0",
}
`;
